controllers/operator/migration/schema/v1/tables.go

```diff
@@ -198,6 +198,17 @@ func ConvertToIdpConfig(idpMap map[string]interface{}, idpConfig *IdpConfig) (er
 	} else {
 		idpMap["enabled"] = true
 	}
+	// SAML with LDAP dependency mongo document will have ldap_config: {ldap_id: <value>}
+	// which directly maps to ldap_id column in sql
+	if ldap_config, ok := idpMap["ldap_config"]; ok {
+		if ldap_config, ok := ldap_config.(map[string]interface{}); ok {
+			// If ldap_config is already a map, fetch the ldap_id
+			ldap_id := ldap_config["ldap_id"]
+			delete(idpMap, "ldap_config") // delete the ldap_config json and introduce ldap_id string
```

Comment:
This deletion isn't necessary - we've removed the json struct tag for `ldap_config` and replaced it with the `ldap_id` struct tag, so `ldap_config` isn't used again after the unmarshal happens.